### PR TITLE
[Sign] CAIP-25

### PR DESCRIPTION
**Pull request by alexander-lsvk**
## Due Dilligence

* [x] Breaking change
* [x] Requires a documentation update

### CODE REVIEW

**Comment by flypaper0** [comment]
Do we also need to change namespaces validation code?
Will be nice to add some tests for optional namespaces in NamespaceValidationTests.swift

**Comment by alexander-lsvk** [comment]
> Do we also need to change namespaces validation code?
> Will be nice to add some tests for optional namespaces in NamespaceValidationTests.swift

Yeah, I mentioned it in Slack, if implementation looks good to you, I'll push some tests 👌🏻

File: Sources/WalletConnectSign/Namespace.swift [hunk 1]
```diff
@@ -47,6 +57,17 @@ enum Namespace {
...
         }
     }
+    
+    static func validate(_ namespaces: [String: OptionalNamespace]) throws {
+        for (_, namespace) in namespaces {
+            if namespace.methods.isEmpty {
+                throw WalletConnectError.unsupportedNamespace(.unsupportedMethods)
+            }
+            if namespace.events.isEmpty {
+                throw WalletConnectError.unsupportedNamespace(.unsupportedEvents)
```
Comment (llbartekll) [inline Sources/WalletConnectSign/Namespace.swift:67]: is it always required to have an event in optional namespace?

Comment (alexander-lsvk) [reply Sources/WalletConnectSign/Namespace.swift:67]: Screenshot 2023-01-30 at 11 38 28
In the doc the whole namespace structure is covered regardless of being `required` or `optional` so I think so.

Comment (llbartekll) [reply Sources/WalletConnectSign/Namespace.swift:67]: should the optional namespace contain the chains too then?
I don't think events and methods cannot be empty but let's discuss it today on the w3w meeting.

**Review by flypaper0 — APPROVED** [review]

**Comment by flypaper0** [comment]
will be good to add a few negative validation tests

**Review by llbartekll — APPROVED** [review]
should we merge it after js/swift/kotlin tests?
there may be some issues discovered

**Review by llbartekll — APPROVED** [review]

**Pull request merged**